code/modules/atmospherics/machinery/unary/vent.dm

```diff
@@ -1,57 +1,65 @@
-
-//TODO: Repath to /unary
-/obj/machinery/atmospherics/pipe/vent
+/obj/machinery/atmospherics/unary/vent
 	icon = 'icons/obj/atmospherics/pipe_vent.dmi'
 	icon_state = "intact"
 	name = "Vent"
 	desc = "A large air vent"
 	level = UNDERFLOOR
-	volume = 250
-	var/obj/machinery/atmospherics/node1
+	plane = PLANE_FLOOR
 
-/obj/machinery/atmospherics/pipe/vent/New()
+/obj/machinery/atmospherics/unary/vent/New()
 	..()
-	initialize_directions = dir
+	src.air_contents.volume = 250
 
-/obj/machinery/atmospherics/pipe/vent/process()
+/obj/machinery/atmospherics/unary/vent/process()
 	..()
-	if(parent)
-		parent.mingle_with_turf(loc, 250)
 
-/obj/machinery/atmospherics/pipe/vent/disposing()
-	node1?.disconnect(src)
-	parent = null
-	..()
+	//stolen from mingle_with_turf WOOHOOH
```

Comment:
Why are we stealing from mingle_with_turf instead of using it directly? What's the difference of this code compared to mingle_with_turf?

Reply:
mingle_with_turf belongs to pipeline, which is a pipe only thing, which means it isn't accessible for nonpipe stuff.

Reply:
there is nothing different except adapting the function to our vent.

Reply:
unfortunate, but i dont feel like changing the current functionality of the vent so i can remove that code paste.

Reply:
Couldn't the relevant parts of the proc be made global and called from both the pipe network's mingle_with_turf and from here?